src/unitxt/eval_utils.py

```diff
@@ -1,29 +1,70 @@
 from functools import singledispatch
-from typing import List
+from typing import List, Optional
 
 import pandas as pd
 
+from .metrics import MetricPipeline
 from .operator import SequentialOperator
 from .stream import MultiStream
 
 
 @singledispatch
-def evaluate(dataset, metric_names: List[str]):
+def evaluate(
+    dataset, metric_names: List[str], compute_conf_intervals: Optional[bool] = False
+):
     """Placeholder for overloading the function, supporting both dataframe input and list input."""
     pass
 
 
 @evaluate.register
-def _(dataset: list, metric_names: List[str]):
+def _(
+    dataset: list,
+    metric_names: List[str],
+    compute_conf_intervals: Optional[bool] = False,
+):
+    global_scores = {}
     for metric_name in metric_names:
         multi_stream = MultiStream.from_iterables({"test": dataset}, copying=True)
         metrics_operator = SequentialOperator(steps=[metric_name])
+
+        if not compute_conf_intervals:
+            first_step = metrics_operator.steps[0]
+            if isinstance(first_step, MetricPipeline):
+                n_samples_before = first_step.metric.n_resamples
+                first_step.metric.n_resamples = None
```

Comment:
perhaps call this `disable_confidence_interval_calculation` (see [here](https://github.com/IBM/unitxt/blob/main/src/unitxt/metrics.py#L77)) on the metric, its the same, but wraps the inner use of the `n_resamples` field.